packages/core/src/browser/preferences/preference-provider.ts

```diff
@@ -47,6 +47,9 @@ export abstract class PreferenceProvider implements Disposable {
     protected readonly onDidPreferencesChangedEmitter = new Emitter<PreferenceProviderDataChanges | undefined>();
     readonly onDidPreferencesChanged: Event<PreferenceProviderDataChanges | undefined> = this.onDidPreferencesChangedEmitter.event;
 
+    protected readonly onDidNotValidPreferencesReadEmitter = new Emitter<any>();
```

Comment:
please provide proper event type and name, for naming, see event naming conventions